commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceDiscountsImporter.java

```diff
@@ -0,0 +1,137 @@
+import com.liferay.portal.kernel.dao.orm.RestrictionsFactoryUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.UserLocalService;
+
+import java.math.BigDecimal;
+
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Steven Smith
+ */
+@Component(service = CommerceDiscountsImporter.class)
+public class CommerceDiscountsImporter {
+
+	public void importCommerceDiscounts(
+			JSONArray jsonArray, long scopeGroupId, long userId)
+		throws Exception {
+
+		User user = _userLocalService.getUser(userId);
+
+		ServiceContext serviceContext = new ServiceContext();
+
+		serviceContext.setScopeGroupId(scopeGroupId);
+		serviceContext.setUserId(userId);
+		serviceContext.setCompanyId(user.getCompanyId());
+
+		for (int i = 0; i < jsonArray.length(); i++) {
+			JSONObject jsonObject = jsonArray.getJSONObject(i);
+
+			_importCommerceDiscount(jsonObject, serviceContext);
+		}
+	}
+
+	private CommerceDiscount _addCommerceDiscount(
+			JSONObject jsonObject, ServiceContext serviceContext)
+		throws PortalException {
+
+		// Add Commerce Discount
+
+		String title = jsonObject.getString("title");
+
+		boolean useCouponCode = jsonObject.getBoolean("useCouponCode");
+		String couponCode = jsonObject.getString("couponCode");
+		boolean usePercentage = jsonObject.getBoolean("usePercentage");
+		BigDecimal maximumDiscountAmount = BigDecimal.valueOf(
+			jsonObject.getDouble("maximumDiscountAmount"));
+		BigDecimal level1 = BigDecimal.valueOf(jsonObject.getDouble("level1"));
+
+		boolean active = jsonObject.getBoolean("active");
+
+		return _commerceDiscountLocalService.addCommerceDiscount(
+			title, CommerceDiscountConstants.TARGET_CATEGORIES, useCouponCode,
+			couponCode, usePercentage, maximumDiscountAmount, level1,
+			BigDecimal.ZERO, BigDecimal.ZERO, BigDecimal.ZERO,
+			CommerceDiscountConstants.LIMITATION_TYPE_UNLIMITED, 0, active, 1,
+			1, 2019, -1, -1, 0, 0, 0, 0, 0, true, serviceContext);
+	}
+
+	private void _importCommerceDiscount(
+			JSONObject jsonObject, ServiceContext serviceContext)
+		throws PortalException {
+
+		CommerceDiscount commerceDiscount = _addCommerceDiscount(
+			jsonObject, serviceContext);
+
+		JSONArray jsonArray = jsonObject.getJSONArray("categories");
+
+		if (jsonArray.length() > 0) {
+			DynamicQuery dynamicQuery =
+				_assetCategoryLocalService.dynamicQuery();
+
+			Criterion criterion = RestrictionsFactoryUtil.eq(
+				"companyId", serviceContext.getCompanyId());
+
+			List<AssetCategory> assetCategories =
+				_assetCategoryLocalService.dynamicQuery(
+					dynamicQuery.add(criterion));
+
+			for (int i = 0; i < jsonArray.length(); i++) {
```

Comment:
@ssmith353 I would prefer a different way, where we do load a category per time, using _assetCategoryLocalService.fetchCategory , so we do avoid an internal loop

Reply:
Hey @marco-leo, 
assetCategoryLocalService does not have a way to fetch the category by the name. 

`fetchCategory(
		long groupId, long parentCategoryId, String name, long vocabularyId)


fetchCategory(
		long categoryId)
`